arches/settings_utils.py

```diff
@@ -1,24 +1,67 @@
-import importlib
 import json
 import os
 import site
 import sys
+from contextlib import contextmanager
 
-from pathlib import Path
+import django
+from django.apps import apps, AppConfig
+from django.conf import global_settings, settings
+
+
+@contextmanager
+def move_to_end_of_sys_path(*paths):
+    _orig_sys_path = sys.path[:]
+    for path in paths:
+        if path in sys.path:
+            sys.path.remove(path)
+            sys.path.append(path)
+    try:
+        yield
+    finally:
+        sys.path = _orig_sys_path
+
+
+def arches_applications():
```

Comment:
nit: `list_arches_applications` or similar name

Reply:
What do you think about starting to use type annotations, at least for function return types -- rather than squeezing this into the name?

Reply:
for me it's not about typing, but naming functions as actions, though typing is never a bad idea 😄